app/src/main/kotlin/com/simplemobiletools/flashlight/dialogs/SleepTimerCustomAlertDialog.kt

```diff
@@ -0,0 +1,133 @@
+                    style = SimpleTheme.typography.headlineSmall,
+                    color = SimpleTheme.colorScheme.onSurface
+                )
+
+                Column(
+                    modifier = Modifier
+                        .padding(
+                            horizontal = SimpleTheme.dimens.margin.extraLarge
+                        )
+                        .padding(
+                            top = SimpleTheme.dimens.margin.extraLarge
+                        )
+                ) {
+                    TextField(
+                        modifier = Modifier.padding(
+                            bottom = SimpleTheme.dimens.margin.large
+                        ),
+                        value = value,
+                        onValueChange = {
+                            value = it.filter { it.isDigit() }
+                        },
+                        label = {
+                            Text(stringResource(id = R.string.value))
+                        },
+                        keyboardOptions = KeyboardOptions(
+                            keyboardType = KeyboardType.Number
+                        )
+                    )
+
+                    items.forEachIndexed { index, item ->
+                        Row(
+                            verticalAlignment = Alignment.CenterVertically
+                        ) {
+                            RadioButton(
+                                selected = index == selectedItem,
+                                onClick = {
+                                    selectedItem = index
+                                }
+                            )
+                            Text(
+                                text = stringResource(id = item)
+                            )
+                        }
+                    }
+                }
+
+                Row(
+                    Modifier.fillMaxWidth(), verticalAlignment = Alignment.CenterVertically,
+                    horizontalArrangement = Arrangement.End
+                ) {
+                    TextButton(onClick = {
+                        onCancelClick?.invoke()
+                        alertDialogState.hide()
+                    }) {
+                        Text(text = stringResource(id = R.string.cancel))
+                    }
+                    TextButton(onClick = {
+                        val enteredValue = Integer.valueOf(value.ifEmpty { "0" })
+                        val multiplier = getMultiplier(items[selectedItem])
+                        onConfirmClick(enteredValue * multiplier)
+                        alertDialogState.hide()
+                    }) {
+                        Text(text = stringResource(id = R.string.ok))
+                    }
+                }
+            }
+        }
+    }
+}
+
+private fun getMultiplier(id: Int) = when (id) {
+    R.string.seconds_raw -> 1
+    R.string.minutes_raw -> 60
```

Comment:
How do you mean? It is needed and used. We need to produce a value in seconds, meaning the choice between `minutes` and `seconds` in the dialog matter. Either way, I decided to make it a bit cleaner, since I didn't like the fact that string resource ID was used in this `when`, so I replaced it with a simple data class.